### PR TITLE
Adding project filters for NodeExport

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description

Adding project filtering for the cfgmgmt.proto NodeExport RPCs.

### :chains: Related Resources

https://github.com/chef/automate/issues/68

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/config-mgmt-service/integration_test/node_export_test.go [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+
+			actualNodeNames := make([]string, 0)
+			jsonparser.ArrayEach(data, func(node []byte, _ jsonparser.ValueType, _ int, err error) {
+				if err == nil {
+					nodeName, err := jsonparser.GetString(node, "name")
+					if err == nil {
+						actualNodeNames = append(actualNodeNames, nodeName)
+					}
+				}
+			})
```
Comment (phiggins) [inline components/config-mgmt-service/integration_test/node_export_test.go]: I'm not sure I understand this part. Do we not have a type for the response returned here?

File: components/config-mgmt-service/integration_test/node_export_test.go [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+					break
+				}
+
+				require.NoError(t, err)
+				data = append(data, tdata.GetContent()...)
+			}
+
+			actualNodeNames := make([]string, 0)
+			jsonparser.ArrayEach(data, func(node []byte, _ jsonparser.ValueType, _ int, err error) {
+				if err == nil {
```
Comment (phiggins) [inline components/config-mgmt-service/integration_test/node_export_test.go]: I know this test is not trying to verify basic correctness of the endpoint, but it feels bad to ignore errors here...

File: components/config-mgmt-service/integration_test/node_export_test.go (hunk 1 above)
Comment (lancewf) [reply components/config-mgmt-service/integration_test/node_export_test.go]: We do not have an exposed type. The type is here https://github.com/chef/automate/blob/master/components/config-mgmt-service/grpcserver/node_export.go#L31. 

I found it easier to just parse the JSON.

File: components/config-mgmt-service/integration_test/node_export_test.go (hunk 2 above)
Comment (lancewf) [reply components/config-mgmt-service/integration_test/node_export_test.go]: That sounds right. I will update this test to fail on a returned error.

**Pull request merged**